### DISCUSSION TITLE
Radeon ProRender for Blender 3.5.0

### DISCUSSION THREAD

**Discussion by bsavery** [GPUOpen-LibrariesAndSDKs/RadeonProRenderBlenderAddon#559] [Announcements]
- Support for Blender 3.3 and Blender 3.4 has been added, including support for the new hair system.
- The RPR Interactive mode has been improved with the following developments:
  - Better support for more shader node types;
  - Even faster rendering with optimized settings out of the box;
  - Hair rendering support;
  - More AOVs supported;
  - Support for OpenColorIO texture spaces;
  - Shadow Catcher support;
  - Per face materials are supported;
  - IES lights are now supported.
- A new Bevel Shader node has been added. This can be used to create small rounded edges on objects. In the physical world, objects rarely have completely sharp edges like polygons. Using this node will enhance the details of artists’ renders.
- Support for the Double Sided node has been added, which allows attaching a different shader to each side of an object mesh.
- The “Pixel Filter” (anti-aliasing) setting for the RPR Final mode has been re-enabled.

<hr /><em>This discussion was created from the release <a href='https://github.com/GPUOpen-LibrariesAndSDKs/RadeonProRenderBlenderAddon/releases/tag/'>Radeon ProRender for Blender 3.5.0</a>.</em>

**Comment by slick3d** [comment]
Not working for me in Linux: when I use the IPR, Blender crashes. The same problem also happens in Houdini. Then, I tried to compile it and I've got compilation error. System: Rocky Linux 9. That's what happens:
![IPR_result](https://user-images.githubusercontent.com/60118168/201449472-5528904e-6847-4672-97fe-17f49874cdac.png)

**Comment by mudapanda2** [comment]
Been trying to use Prorender for two years now and just haven't been down with it, using an old weaksauce R9 390 8gb. No idea what you did with this version but it's making breakfast out of demos faster than I've ever experienced. Good job, if the dang driver support for this GPU wasn't so bad I'd like to try Prorender on Linux again. Good job on this.

**Comment by FireAngelOfLondon** [comment]
First impressions are that it is faster, but there is a stupidly long delay when pressing ESCape to abort a render.  It can take minutes to respond to this, which is annoying.  It does start rendering very quickly now rather than taking a silly time to compile shaders, but I would still like to see it abort a render in a reasonable time.  It's actually faster to force quit Blender from outside and then reload the file you were working on.  This is irritating and really silly.

**Reply by FireAngelOfLondon** [reply]
Oh, and I forgot; any attempt to use an Open EXR file as an IBL and Blender crashes as soon as rendering starts.

I'm on a Mac Pro, 128GB RAM, Catalina OS, Radeon RX580 8Gb.

**Comment by DionXein** [comment]
Great work! Render become more stable and faster then ever. You did very hard work, and we appreciate it! And thank you!

**Comment by alexwilderuk** [comment]
Hey team,
I'm getting the following error when I try to Import Material: 

<img width="852" alt="Screenshot 2023-02-04 at 13 35 42" src="https://user-images.githubusercontent.com/124445248/216771014-f97d4eb0-2901-4561-b0cd-8d18483f6a3b.png">

(Strangely 'Diamond' worked. 'Rubber' and 'Gold' gave similar errors)
Any ideas? Thanks in advance.

**Reply by alexwilderuk** [reply]
Solved. Apologies - I was trying to import when another item already had a material.

**Comment by TerydiyRadymor** [comment]
І куди цю хуйню кидати чи як встановити, що блять дальше???
and how this shit instal? or where drop it, how that shit must work???

**Reply by bsavery** [reply]
https://radeon-pro.github.io/RadeonProRenderDocs/en/plugins/blender/installation.html

**Reply by TerydiyRadymor** [reply]
too late (why is there no description under the file), I found an installer with a complete reinstallation of the driver, (amd became bad І download the driver from the official website and they are missing something, those developers are sick, if not for Blender, then Vegas does not work)

**Reply by FireAngelOfLondon** [reply]
> І куди цю хуйню кидати чи як встановити, що блять дальше??? and how this shit instal? or where drop it, how that shit must work???

If you want real answers to real questions it would be far more productive to ask politely.  Being blunt is one thing, being rude is not likely to result in getting help as quickly or as readily.  I am slightly amazed you got even the helpful link above.

**Reply by TerydiyRadymor** [reply]
imagine that I am shocked by the fact that I have to look for some applications and drivers that are not on the official sites, then I am guilty of AMD's crookedness, and maybe I should replace the writing (in more detail here), they must writed the installation is here

**Reply by bsavery** [reply]
Hey @TerydiyRadymor, not sure at all what you're trying to do.  This is a plugin for blender, not at all related to drivers.  Good luck.

**Reply by TerydiyRadymor** [reply]
thanks for the luck. I installed some amd video card drivers, I don't even know on which site I found them, and everything worked.

**Comment by ihateedensgarden** [comment]
Hi! Trying to install the addon on Mint 21. When I try to enable the addon, blender just crashes. This doesn't seem to be a known issue afaik. If you guys could look into this it would be great :D

**Comment by sergofc** [comment]
So guys how does one access this new Bevel Shader node. Am I missing something? Can't seem to find any documentation about it either.
All I see is the blender's bevel node in the shader editor, which as supposed does not work. Any help?

**Reply by sergofc** [reply]
Ok I found out it works only in final render, and not in viewport

**Reply by bsavery** [reply]
If you set the viewport mode to "RPR - Final" in the settings it should work.  We're looking to add support to interactive mode in the future.

**Comment by chrisx2ds** [comment]
the source code is here https://github.com/GPUOpen-LibrariesAndSDKs/RadeonProRenderBlenderAddon/releases/tag/v3.5.2

can anyone help to compile it for windows and upload again?
 
Thanks

it used to be there, but the file is gone, and only source code there


![3 5 2](https://github.com/GPUOpen-LibrariesAndSDKs/RadeonProRenderBlenderAddon/assets/29679232/7855a6d6-1076-42e0-97d0-decafd74d4b8)

**Comment by hbecerra81** [comment]
Hey,
So, I have this older workstation I'm trying to test this on atm, which is running on Kubuntu 24.04 LTS. The GPU in question/use is an AMD ATI FirePro V5900, which is by default running under open-source MESA video drivers (I'm also assuming this is the culprit too) since it's a clean install and it was way too confusing trying to install the older proprietary drivers in Linux os's to get better functionality from this GPU. Under Windows 10/11os, I've managed to use this add-on perfectly fine in Blender 3.6.15 LTS, but, when I try to enable it under this Linux setup, I get trace errors as follows:
```
Traceback (most recent call last):
  File "/opt/blender/3.6/scripts/modules/addon_utils.py", line 333, in enable
    mod = __import__(module_name)
  File "/home/hugob/.config/blender/3.6/scripts/addons/rprblender/__init__.py", line 52, in <module>
    from .engine.viewport_engine import ViewportEngine
  File "/home/hugob/.config/blender/3.6/scripts/addons/rprblender/engine/viewport_engine.py", line 32, in <module>
    from rprblender.utils import gl
  File "/home/hugob/.config/blender/3.6/scripts/addons/rprblender/utils/gl.py", line 25, in <module>
    gl = cdll.LoadLibrary('libGL.so')
  File "/opt/blender/3.6/python/lib/python3.10/ctypes/__init__.py", line 452, in LoadLibrary
    return self._dlltype(name)
  File "/opt/blender/3.6/python/lib/python3.10/ctypes/__init__.py", line 374, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: libGL.so: cannot open shared object file: No such file or directory
```
Since at first, I was trying to run this in the snap version of Blender, I thought it was the restrictive sandboxing environment that was causing the issue with the add-on not being able to locate the libGL.so directory which was outside of the snap environment. Then, I tried it on a normal install using a Blender tar file directly from their official sources. Yet again, I was met with the same resulting trace error. Making it clear that it wasn't an issue with not having access to a set directory but rather just possibly not having the right type of drivers in place to allow Blender to enable the add-on as it should, correct?

I do understand that this is more of an issue about my PC's GPU drivers rather than y'all's add-on, but is there any way you could do me a solid on this as far as how to get this working in Linux (should the issue not be a driver since the open-source drivers are newer than the old proprietary ones)? I was able to use Cycles with GPU selected, which is why I thought this might be a better way to go currently rather than using old drivers in a newer OS. I really appreciate any and all feedback you can give me on this issue/topic.

**Reply by G4br13lG0** [reply]
Hi hbcerra81, I believe is possible to use that gpu with newer drivers if you change the config to use opencl (under render tab then about tab right on bottom of render tab, if I am not mistaken)

Now about libGl.so, if I also recall correctly is a library that is installed with legacy pro drivers (also proprietary), and I need to check but  I believe  driver 22.20.5 is the last version that officially support it
